### PR TITLE
Centralize handle-based post-object creation code.

### PR DESCRIPTION
**Pull request by jaykrell**
i.e. merge 20e69fa1996e32bba66cea4a723b3da0ee913243 and d7626e8217f3d2b8dbcabfa8281475fe8b2eb851.

Remove some redundant error_init calls.

### CODE REVIEW

File: mono/metadata/object.c [hunk 1]
```diff
@@ -5391,8 +5390,10 @@ mono_object_new_common_tail (MonoObject* o, MonoClass* klass, MonoError* error)
 static MonoObjectHandle
```
Comment (vargaz) [inline mono/metadata/object.c]: In most c coding conventions including ours,. the '*' is next to the variable name, not the type.

File: mono/metadata/object.c [hunk 2]
```diff
@@ -5381,7 +5381,6 @@ mono_object_new_common_tail (MonoObject* o, MonoClass* klass, MonoError* error)
 	return o;
 }
 
-#if 0 // FIXMEcoop awaiting https://github.com/mono/mono/pull/6876
 /**
  * mono_object_new_handle_tail:
  *
```
Comment (vargaz) [inline mono/metadata/object.c]: Static functions don't need a mono_ prefix.

Comment (jaykrell) [reply mono/metadata/object.c]: Removed on this and the non-handle on.

File: mono/metadata/object.c (hunk 1 above)
Comment (jaykrell) [reply mono/metadata/object.c]: Fixed a few.

**Pull request merged**